core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestUtils.java

```diff
@@ -236,10 +247,18 @@ public static S3Exception toBucketS3Exception(Exception exception, String resour
    *
    * @param exception Exception thrown when process s3 object rest request
    * @param resource object complete path
+   * @param auditContext the audit context for exception
    * @return instance of {@link S3Exception}
    */
-  public static S3Exception toObjectS3Exception(Exception exception, String resource) {
+  public static S3Exception toObjectS3Exception(Exception exception, String resource,
+                                                @Nullable AuditContext auditContext) {
```

Comment:
I would prefer if you overloaded the method definition and avoided refactoring the callers to this method. It looks like there are only 3 calls to this method where `auditContext != null`.